core/src/main/scala/akka/persistence/r2dbc/internal/BySliceQuery.scala

```diff
@@ -416,20 +416,28 @@ import org.slf4j.Logger
       state.backtracking && state.rowCount < settings.querySettings.bufferSize - state.backtrackingExpectFiltered
     }
 
+    def switchToBacktracking(state: QueryState, newIdleCount: Long): Boolean = {
+      // Note that when starting the query with offset = NoOffset it will switch to backtracking immediately after
+      // the first normal query because between(latestBacktracking.timestamp, latest.timestamp) > halfBacktrackingWindow
+
+      val qSettings = settings.querySettings
+      import akka.util.JavaDurationConverters._
+
+      qSettings.backtrackingEnabled &&
+      !state.backtracking &&
+      state.latest != TimestampOffset.Zero &&
+      state.latest.timestamp.isAfter(clock.instant().minus(qSettings.backtrackingWindow.asJava)) &&
```

Comment:
Have to think about the exact timestamps for this. One candidate is to look at the `toTimestamp`, but as the code is structured now that is kind calculated afterwards.

Reply:
must also adjust the `latestBacktracking` because otherwise when backtracking is activated again it would start from the far behind latestBacktracking